### PR TITLE
Tables by Player Count

### PR DESCRIPTION
**Pull request by elibillinger**

### CODE REVIEW

File: src/client/modules/my/resultsFetcher/resultsFetcher.ts [hunk 1]
```diff
@@ -43,10 +44,33 @@ export function getRawResults(): Promise<GameResultsData[]> {
...
+        } else {
+            accum[game_label]++;
+        }
+        return accum;
+        
+    }, {});
+    this.gameSize = Object.entries(playersPerGame).map(([key, value]) => { return {game_label: key, player_num: value}});
+    let rawDataFiltered: GameResultsData[] = []
+    // if game size is between 2-6 filter for only games of that player size else return all data
+    if(game_size >= 2 && game_size <=6){
```
Comment (gik0geck0) [inline src/client/modules/my/resultsFetcher/resultsFetcher.ts:65]: Special pivots like this are an anti-pattern. There's no correlation between game_size=1 and the outcome. This should be a new method (with sub-methods for shared processing)

File: src/server/db_setup.ts [hunk 2]
```diff
@@ -56,8 +56,6 @@ async function init(): Promise<void> {
             csvData.push(data);
         })
         .on("end", function() {
-            // remove the first line: header
-            csvData.shift();
```
Comment (gik0geck0) [inline src/server/db_setup.ts:60]: What changed such that this isn't needed? Does fast-csv skip/understand that the first line is usually a header, so this basically changes nothing?

Comment (elibillinger) [reply src/server/db_setup.ts:60]: The csvData.shift() was removing the first line of the GameLogs.csv file. This line was info for one of the players in the first game - it was (20200911a,1,Shelby,53). This was messing up the data for two player and three player games. Removing this line fixed that issue.

File: src/client/modules/my/resultsFetcher/resultsFetcher.ts [hunk 3]
```diff
@@ -43,8 +44,86 @@ export function getRawResults(): Promise<GameResultsData[]> {
...
+    calculatePlayerStats(PlayerStats,resultsData);
+   return PlayerStats;
+}
+
+export function extractTwoPlayerStats (resultsData: GameResultsData[]) {
+    let PlayerStats: PlayerStatsAllGames[] = [];
+    let gameSize: PlayersPerGame[] = [];
+    let rawDataFiltered: GameResultsData[] = [];
+    gameSize = getGameSizes(resultsData);
+    let filteredGameSize = this.gameSize.filter(game => game.player_num === 2);
```
Comment (gik0geck0) [inline src/client/modules/my/resultsFetcher/resultsFetcher.ts:59]: We could collapse 2-6 player stats functions back together where the number of players is a parameter. It was just All vs N-players that I wanted you to split.

I really like how these all just do a filter before calculatePlayerStats; it makes it really simple and composable; just change the data-set to collect stats over, and get that same collection of stats.

**Pull request merged**